Clustering/Hdbscan/src/main/java/org/tribuo/clustering/hdbscan/HdbscanTrainer.java

```diff
@@ -117,28 +135,61 @@ private HdbscanTrainer() {
      * Constructs an HDBSCAN* trainer with only the minClusterSize parameter.
      *
      * @param minClusterSize The minimum number of points required to form a cluster.
-     * {@link #distanceType} defaults to {@link Distance#EUCLIDEAN}, {@link #k} defaults to {@link #minClusterSize},
+     * {@link #distType} defaults to {@link DistanceType#L2}, {@link #k} defaults to {@link #minClusterSize},
      * and {@link #numThreads} defaults to 1.
      */
     public HdbscanTrainer(int minClusterSize) {
-        this(minClusterSize, Distance.EUCLIDEAN, minClusterSize, 1);
+        this(minClusterSize, DistanceType.L2, minClusterSize, 1);
     }
 
     /**
      * Constructs an HDBSCAN* trainer using the supplied parameters.
+     * @deprecated
+     * This Constructor is deprecated in version 4.3.
      *
      * @param minClusterSize The minimum number of points required to form a cluster.
      * @param distanceType The distance function.
      * @param k The number of nearest-neighbors to use in the initial density approximation.
      * @param numThreads The number of threads.
      */
+    @Deprecated
     public HdbscanTrainer(int minClusterSize, Distance distanceType, int k, int numThreads) {
         this.minClusterSize = minClusterSize;
-        this.distanceType = distanceType;
+        this.distType = distanceType.getDistanceType();
+        this.k = k;
+        this.numThreads = numThreads;
```

Comment:
Similar to the other constructors, maybe this should directly call the `HdbscanTrainer(int,DistanceType,int,int)` constructor?